### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by machacekondra**
Signed-off-by: Ondra Machacek <omachace@redhat.com>

### CODE REVIEW

File: cluster-sync/imageio/imageio.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      app: imageio
+  template:
+    metadata:
+      labels:
+        app: imageio
+    spec:
+      serviceAccountName: cdi-sa
+      containers:
+      - name: imageiotest
+        image: machacekondra/imageiotest:test
```
Comment (pkliczewski) [inline cluster-sync/imageio/imageio.yaml:20]: What about Adam's comment to use kubevirt org for the tests containers?

File: cluster-sync/clean.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
+#!/bin/bash -e
+  
+source ./hack/build/config.sh
+source ./cluster-up/hack/common.sh
+source ./cluster-up/cluster/${KUBEVIRT_PROVIDER}/provider.sh
+
+echo "Cleaning up ..."
```
Comment (pkliczewski) [inline cluster-sync/clean.sh:7]: It seems to be cdi specific. As one of the next steps we can add kubevirt cleanup as well.

File: cluster-sync/k8s-1.16.2/ceph_sc.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+apiVersion: storage.k8s.io/v1
```
Comment (pkliczewski) [inline cluster-sync/k8s-1.16.2/ceph_sc.yaml]: I thought that we wanted to use nfs as listed in ephemeral provider. I am not sure whether we need to add more storageclasses.

File: cluster-sync/okd-4.3/provider.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    echo "Relabeling SELinux on master $master_name"
+    ./cluster-up/ssh.sh $master_name "sudo chcon -t container_file_t -R /var/hpvolumes"
+  else
+    echo "Using local volume storage"
+    #Make sure local is not default
+    _kubectl patch storageclass local -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
+  fi
+}
+
+function configure_ceph() {
```
Comment (pkliczewski) [inline cluster-sync/okd-4.3/provider.sh]: Maybe we could remove ceph configuration here as well.

**Review by pkliczewski — COMMENTED** [review]
As for WIP PR it looks great so far :)

File: cluster-sync/imageio/imageio.yaml (hunk 1 above)
Comment (machacekondra) [reply cluster-sync/imageio/imageio.yaml:20]: Well, that was one option. Second was to make it public, which I did:

 - https://github.com/machacekondra/imageiotest
 - https://github.com/machacekondra/fakeovirt

File: cluster-sync/k8s-1.16.2/ceph_sc.yaml (hunk 3 above)
Comment (machacekondra) [reply cluster-sync/k8s-1.16.2/ceph_sc.yaml]: Right, I'll remove others.

File: cluster-sync/external/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Container Image Registry, we tie into the existing Make/Build workflow, but modify things
+such that:
+1. We build container images and push to the specified registry (instead of `registry:5000`)
+2. We generate the manifests with the provided DOCKER_PREFIX (uses default port)
+3. Uses your configured `kubectl` and deploys your build to the existing cluster
+
+# Building images and pushing to your registry
+
+```bash
+export DOCKER_PREFIX=index.docker.io/barney_rubble
```
Comment (pkliczewski) [inline cluster-sync/external/README.md:13]: What is this `DOCKER_PREFIX`. I get 404 when using it in the browser.

File: cluster-sync/sync.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+install_kubevirt
+
+configure_storage
+
+# We skip the functional test additions for external provider for now, as they're specific
+if [ "${KUBEVIRT_PROVIDER}" != "external" ]; then
+  _kubectl apply -f "./cluster-sync/imageio/imageio.yaml"
+  _kubectl wait deploy/imageio-deployment -n cdi --for=condition=Available --timeout=480s
+fi
+
```
Comment (pkliczewski) [inline cluster-sync/sync.sh:27]: Do we want to install `cluster-network-addons-operator ` here?

File: cluster-sync/sync.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash -e
+
+export CDI_INSTALL_TIMEOUT=${CDI_INSTALL_TIMEOUT:-120}
+
+source ./cluster-up/hack/common.sh
+source ./cluster-up/cluster/${KUBEVIRT_PROVIDER}/provider.sh
+source ./cluster-sync/${KUBEVIRT_PROVIDER}/provider.sh
+
+# Install CDI
+install_cdi
```
Comment (pkliczewski) [inline cluster-sync/sync.sh:10]: Maybe we could move cdi install to a function like we have for `kubevirt`.

**Review by jakub-dzon — APPROVED** [review]

File: cluster-sync/sync.sh (hunk 6 above)
Comment (machacekondra) [reply cluster-sync/sync.sh:27]: When we start tests multus, then definitelly.

File: cluster-sync/sync.sh (hunk 7 above)
Comment (machacekondra) [reply cluster-sync/sync.sh:10]: Well, it is a function same as kubevirt. You mean to add waiting to the function?

File: cluster-sync/external/README.md (hunk 5 above)
Comment (machacekondra) [reply cluster-sync/external/README.md:13]: It's just an example how to build the images for your own.

File: cluster-sync/sync.sh (hunk 7 above)
Comment (pkliczewski) [reply cluster-sync/sync.sh:10]: yeah, I meant everything related to CDI

File: cluster-sync/external/README.md (hunk 5 above)
Comment (pkliczewski) [reply cluster-sync/external/README.md:13]: OK

**Review by pkliczewski — APPROVED** [review]
It is OK to have it as is. We can improve in next PR.

**Pull request merged**